File: cardbattlemodule/src/main/java/com/syahputrareno975/cardbattlemodule/model/room/RoomDataModel.java
package com.syahputrareno975.cardbattlemodule.model.room;

import com.syahputrareno975.cardbattlemodule.model.playerWithCard.PlayerWithCardsModel;
import com.syahputrareno975.cardbattlemodule.model.roomReward.RoomRewardModel;
import java.io.Serializable;
import java.util.ArrayList;

public class RoomDataModel implements Serializable {

    public String Id = "";
    public String RoomName = "";
    public ArrayList<PlayerWithCardsModel> Players = new ArrayList<PlayerWithCardsModel>();
    public int MaxPlayer = 0;
    public int MaxPlayerDeck = 0;
    public int  MaxCurrentDeployment = 0;
    public int MaxDeploment = 0;
    public Long EachPlayerHealth = 0L;
    public int CoolDownTime = 0;
    public RoomRewardModel Reward = new RoomRewardModel();

    public RoomDataModel() {
    }

    public RoomDataModel(String id, String roomName, ArrayList<PlayerWithCardsModel> players, int maxPlayer, int maxPlayerDeck, int maxCurrentDeployment, int maxDeploment, Long eachPlayerHealth, int coolDownTime, RoomRewardModel reward) {
        Id = id;
        RoomName = roomName;
        Players = players;
        MaxPlayer = maxPlayer;
        MaxPlayerDeck = maxPlayerDeck;
        MaxCurrentDeployment = maxCurrentDeployment;
        MaxDeploment = maxDeploment;
        EachPlayerHealth = eachPlayerHealth;
        CoolDownTime = coolDownTime;
        Reward = reward;
    }
}

File: app/src/main/java/com/syahputrareno975/simplecardbattle/model/room/RoomDataModel.kt
package com.syahputrareno975.simplecardbattle.model.room

import com.syahputrareno975.simplecardbattle.model.card.CardModel
import com.syahputrareno975.simplecardbattle.model.playerWithCard.PlayerWithCardsModel
import com.syahputrareno975.simplecardbattle.model.roomReward.RoomRewardModel
import java.io.Serializable

class RoomDataModel : Serializable {
    var Id = ""
    var RoomName = ""
    var Players = ArrayList<PlayerWithCardsModel>()
    var MaxPlayer = 0
    var MaxPlayerDeck = 0
    var MaxCurrentDeployment = 0
    var MaxDeploment = 0
    var EachPlayerHealth : Long = 0
    var CoolDownTime = 0
    var Reward : RoomRewardModel = RoomRewardModel()


    constructor()
    constructor(
        id: String,
        RoomName: String,
        Players: ArrayList<PlayerWithCardsModel>,
        MaxPlayer: Int,
        MaxPlayerDeck: Int,
        maxCurrentDeployment : Int,
        MaxDeploment: Int,
        EachPlayerHealth: Long,
        CoolDownTime: Int,
        reward : RoomRewardModel
    ) {
        this.Id = id
        this.RoomName = RoomName
        this.Players = Players
        this.MaxPlayer = MaxPlayer
        this.MaxPlayerDeck = MaxPlayerDeck
        this.MaxDeploment = MaxDeploment
        this.EachPlayerHealth = EachPlayerHealth
        this.CoolDownTime = CoolDownTime
        this.Reward = reward
        this.MaxCurrentDeployment = maxCurrentDeployment
    }

}


